staging/src/k8s.io/client-go/util/certificate/certificate_manager.go

```diff
@@ -115,6 +115,9 @@ type Config struct {
 	// CertificateExpiration will record a metric that shows the remaining
 	// lifetime of the certificate.
 	CertificateExpiration Gauge
+	// CertificateRotation will record a metric showing the time in seconds
+	// that certificates lived before being rotated.
+	CertificateRotation Histogram
```

Comment:
Yeah so...
CertificateExpiration is a Gauge because it is like reading a static status. When does the kubelet's serving cert expire? "June 6th 2020" or something,  and that reading will be the same until its eventually rotated and reads something like "March 16th 2021" or some other future date. There isn't really a benefit in treating it like a history of values or a graph. E.g. ["June 6th 2020", "March 16th 2010", "..."] in some big pointless histogram.

For Certificate Rotation we do care about the history and trends. In efforts of lowering the TTL and increasing the rotation cadence of a cluster's certificates one can track certificate lifetimes when rotated:
["cert lived for 250 days", "cert lived for 209 days", "cert lived for 244 days", ...]
Tooling can then be setup to watch for cert rotations increasing or decreasing unexpected.

Reply:
Gotcha! That's definitely a helpful explanation. Thoughts on including what you wrote as comments for both `CertificateExpiration` and `CertificateRotation`? At least for me, it was non-obvious, so a comment about the "why" feels appropriate.

Reply:
Done.